### PR TITLE
Save template thumbnails and resources to an AGOL item

### PR DESCRIPTION
**Pull request by MikeTschudi**
Working on its tests now; wanted to keep PR shorter

### CODE REVIEW

File: src/itemTypes/common.ts [hunk 1]
```diff
@@ -214,7 +220,7 @@ export function updateItemData (
       updateResp => {
         resolve(id);
       },
-      () => reject()
+      () => reject({ success: false })
```
Comment (jmhauck) [inline src/itemTypes/common.ts]: Maybe I'm missing it from just seeing this little bit...why is this only rejecting with `{ success: false }`

Comment (MikeTschudi) [reply src/itemTypes/common.ts]: Just hadn't gotten to updating the all of the rejects to the new form

File: test/mocks/agolItems.ts [hunk 2]
```diff
@@ -714,7 +724,13 @@ export function createAGOLRelationship (
 export function getAnImageResponse (
 ): any {
   const fs = require("fs");
-  return fs.createReadStream("./test/mocks/success.png");
+  if (fs.createReadStream) {
+    // Node test
+    return fs.createReadStream("./test/mocks/success.png");
+  } else {
+    // Chrome test
+    return new Blob([atob(imageAsDataUri(false))], {type: "image/png"})
```
Comment (jmhauck) [inline test/mocks/agolItems.ts]: minor issue but missing a `;`

Comment (MikeTschudi) [reply test/mocks/agolItems.ts]: Thank you; I probably should add linting to the tests

**Pull request merged**